### DISCUSSION TITLE
Ignition Dataset datatype and node-red

### DISCUSSION THREAD

**Discussion by Theodorska1** [thomassorensen2650/node-red-contrib-mqtt-sparkplug-plus#29] [Q&A]
Does anyone know if it is possible to send and recieve Ignition dataset with these nodes?
From my understanding it is only possible to send datatypes according to the [specification ](https://s3.amazonaws.com/ignition-modules/Current/Sparkplug+Specification.pdf): 
•DOUBLE
• FLOAT
• INT64
• INT32
• BOOL
• STRING
• BYTES
or unknown

However, if I want to send an array, json, or dataset, I wonder if i can use datatype "unknown" and structure the payload in a way that would be recognizable in Ignition as a dataset?

**Comment by Theodorska1** [comment]
We solved this by passing the array as a string and converting it

**Comment by thomassorensen2650** [comment]
This is a pretty useful feature. I have added support for datasets in datasupport branch,  I just need to add some input validation. I will see if i can get this merged in early next year.

**Comment by thomassorensen2650** [comment]
Added support in PR #36 will be added in next release
